sdks/java/maven-archetypes/examples/pom.xml

```diff
@@ -49,6 +49,21 @@
         <plugin>
           <artifactId>maven-archetype-plugin</artifactId>
           <version>2.4</version>
+          <executions>
+            <!-- archetype-packaging above binds this plugin's goals as follows:
+                     archetype:jar to the package phase,
+                     archetype:integration-test to the integration-test phase,
+                     archetype:update-local-catalog to the install phase.
+                 We defer the integration-test goal to the install phase, since
+                 this test actually depends on the core SDK to be installed. -->
```

Comment:
Well, not quite. Ideally, the build should be always be using an artifact produced by this build. Say, we could have introduced backward-incompatible changes that would break this again even if a matching artifact is released previously.